src/axom/core/Array.hpp

```diff
@@ -494,7 +500,7 @@ class Array : public ArrayBase<T, DIM, Array<T, DIM>>
   /*!
    * \brief Exchanges the contents of this Array with the other.
    */
-  void swap(Array<T, DIM>& other);
+  void swap(Array<T, DIM, SPACE>& other);
```

Comment:
Can this be extended to accept `Array other` from other memory spaces as well?

Reply:
I can add a swap function between memory spaces but I'd be curious to hear what some use cases might be for this - in the general case you'd have to copy both sets of data between memory spaces, so I think the only thing you'd really be swapping are the housekeeping/metadata variables.

Reply:
It might be useful for memory spaces that are compatible.
E.g. a `Dynamic` space that happens to be device memory (via umpire) and a `Device` space ?